test/scales/linearScaleTests.ts

```diff
@@ -1,134 +1,267 @@
-      assert.deepEqual(scale.domain(), [desiredMin, desiredMax], "setting domainMin() and domainMax() sets the domain");
-
-      scale.autoDomain();
-      let bigMin = 10;
-      let smallMax = -10;
-      scale.domainMin(bigMin);
-      scale.domainMax(smallMax);
-      assert.deepEqual(scale.domain(), [bigMin, smallMax], "setting both is allowed even if it reverse the domain");
+      it("filters out invalid numbers when using extentOfValues()", () => {
+        let arrayWithBadValues: any[] = [null, NaN, undefined, Infinity, -Infinity, "a string", 1, 1.2];
+        let expectedExtent = [1, 1.2];
+        let extent = scale.extentOfValues(arrayWithBadValues);
+        assert.deepEqual(extent, expectedExtent, "invalid values were filtered out");
+      });
+
+      it("does not accept NaN or Infinity as domain", () => {
+        let warningCalled = false;
+        let oldWarn = Plottable.Utils.Window.warn;
+        Plottable.Utils.Window.warn = (msg: string) => {
+          if (msg.indexOf("NaN or Infinity") > -1) {
+            warningCalled = true;
+          }
+        };
+
+        scale.domain([1, 2]);
+        assert.deepEqual(scale.domain(), [1, 2], "initial domain is set");
+
+        warningCalled = false;
+        scale.domain([5, Infinity]);
+        assert.deepEqual(scale.domain(), [1, 2], "Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to Infinity");
+
+        warningCalled = false;
+        scale.domain([-Infinity, 5]);
+        assert.deepEqual(scale.domain(), [1, 2], "-Infinity containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to -Infinity");
+
+        warningCalled = false;
+        scale.domain([NaN, 7]);
+        assert.deepEqual(scale.domain(), [1, 2], "NaN containing domain was ignored");
+        assert.isTrue(warningCalled, "a warning was thrown for setting domain to NaN");
+
+        warningCalled = false;
+        scale.domain([-1, 5]);
+        assert.deepEqual(scale.domain(), [-1, 5], "Regular domains still accepted");
+        assert.isFalse(warningCalled, "a warning is not thrown when correct domains are given");
+
+        Plottable.Utils.Window.warn = oldWarn;
+      });
+
+      it("accepts NaN or Infinity as range", () => {
+        scale.range([5, Infinity]);
+        assert.deepEqual(scale.range(), [5, Infinity], "Infinity accepted as part of range");
+        scale.range([-Infinity, 5]);
+        assert.deepEqual(scale.range(), [-Infinity, 5], "-Infinity accepted as part of range");
+        scale.range([NaN, 7]);
+        assert.deepEqual(scale.range(), [NaN, 7], "NaN accepted as part of range");
+      });
+
+      it("can receive custom tick generator", () => {
+        scale.domain([0, 10]);
+        let defaultTicks = scale.ticks();
+        assert.strictEqual(defaultTicks.length, 11, "ticks were generated correctly with default generator");
```

Comment:
Do we want to assert that specific values were generated, as in the later `assert()`?